sh.c

```diff
@@ -1769,17 +1769,63 @@ void comp_body(ast node) {
 
 // Return if the statement is a break or return statement, meaning that the block should be terminated.
 // A switch conditional block is considered terminated if it ends with a break or return statement.
-bool comp_switch_block_statement(ast node, bool start_in_tail_position) {
+bool comp_switch_block_statement(ast node, bool else_if, bool start_in_tail_position) {
+  ast statement = 0;
+  bool termination_lhs = false;
+  bool termination_rhs = false;
+
   if (get_op(node) == BREAK_KW) {
+    append_glo_decl(wrap_char(':'));
     return true;
   } else if (get_op(node) == RETURN_KW) {
     // A return marks the end of the conditional block, so it's in tail position
     in_tail_position = start_in_tail_position;
     comp_statement(node, false);
     return true;
   } else if (get_op(node) == CASE_KW || get_op(node) == DEFAULT_KW) {
-    fatal_error("comp_statement: case must be at the beginning of a switch block, and each block must end with a break or return statement");
+    fatal_error("comp_switch_block_statement: case must be at the beginning of a switch block, and each block must end with a break or return statement");
+    return false;
+  } else if (get_op(node) == '{') {
+    statement = get_child(node, 0);
+    while(statement != 0) {
```

Comment:
I don't fully remember how `{` AST nodes are represented but this should be the same to how it's done in `comp_body`